contracts/ERC20MultiDelegate.sol

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.2;
+
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Votes.sol";
+
+/**
+ * @dev A child contract which will be deployed by the ERC20MultiDelegate utility contract
+ * This is a proxy delegator contract to vote given delegatee on behalf of original delegator
+ */
+contract ERC20ProxyDelegator {
+    constructor(ERC20Votes _token, address _delegatee) {
+        _token.approve(msg.sender, type(uint256).max);
+        _token.delegate(_delegatee);
+    }
+}
+
+/**
+ * @dev A utility contract to let delegators to pick multiple delegatee
+ */
+contract ERC20MultiDelegate is ERC1155, Ownable {
+    using Address for address;
+
+    ERC20Votes token;
+
+    /**
+     * @dev Constructor.
+     * @param _token The ERC20 token address
+     */
+    constructor(ERC20Votes _token) ERC1155("http://some.metadata.url/{id}") {
+        token = _token;
+    }
+
+    /**
+     * @dev Public method for the delegation of multiple delegatees.
+     * @param delegatees List of delegatee addresses
+     * @param amounts ERC20 voting power amount to be distributed among delegatees
+     */
+    function delegateMulti(
+        address[] calldata delegatees,
+        uint256[] calldata amounts
+    ) external {
+        require(
+            delegatees.length > 0,
+            "You should pick at least one delegatee"
+        );
+        require(
+            delegatees.length == amounts.length,
+            "Amounts should be defined for each delegatee"
+        );
+
+        uint256[] memory ids = new uint256[](delegatees.length);
+
+        for (uint256 index = 0; index < delegatees.length; ) {
+            address delegatee = delegatees[index];
+            // creates a proxy delegator contract with deterministic address
+            // salt occurs from delegatee address + sender address
+            bytes memory bytecode = getBytecode(token, delegatee);
```

Comment:
`getBytecode` and `getAddress` should probably be combined into a single utility function.